applications/FemToDemApplication/custom_elements/femdem3d_element.cpp

```diff
@@ -1010,97 +1030,101 @@ double FemDem3DElement::GetMaxValue(Vector Strain)
 	return V[n - 1];
 }
 
-double FemDem3DElement::GetMaxAbsValue(Vector Strain)
+double FemDem3DElement::GetMaxAbsValue(
+	const Vector& rArrayValues
+	)
 {
-	Vector V;
-	int n = Strain.size();
-	V.resize(n);
+	const std::size_t dimension = rArrayValues.size();
+	std::vector<double> non_zero_values;
 
-	for (int cont = 0; cont < n; cont++) {
-		V[cont] = std::abs(Strain[cont]);
+	for (std::size_t i = 1; i < dimension; ++i) {
+		if (std::abs(rArrayValues[i]) > tolerance)
+			non_zero_values.push_back(std::abs(rArrayValues[i]));
 	}
-	int imin = 0;
+	KRATOS_ERROR_IF(non_zero_values.size() == 0) << "The strain vector is full of 0's..." << std::endl;
 
-	for (unsigned int i = 0; i < n; i++) {
-		for (unsigned int j = 0; j < n - 1; j++) {
-			if (V[j] > V[j + 1]) {
-				double aux = V[j];
-				V[j] = V[j + 1];
-				V[j + 1] = aux;
-			}
-		}
+	double aux = std::abs(non_zero_values[0]);
+	for (std::size_t i = 1; i < non_zero_values.size(); ++i) {
+		if (non_zero_values[i] > aux)
+			aux = non_zero_values[i];
 	}
-	return V[n - 1];
+	return aux;
 }
 
-double FemDem3DElement::GetMinAbsValue(Vector Strain)
+double FemDem3DElement::GetMinAbsValue(

```

Comment:
This is not very efficient, look at the StructuralMechanics implementation

Reply:
do you mean the one implemented in the tangent calculator?

thx

Reply:
Yep

Reply:
done, thx